### PR TITLE
ramips: add support for BOLT! Arion

### PR DESCRIPTION
**Pull request by kopijahe**
This device is from now-defunct BOLT! ISP in Indonesia.
The original firmware is based on mediatek SDK running linux 2.6 or 3.x in later revision.

Specifications:

- SoC:      MediaTek MT7621
- Flash:    32 MiB NOR SPI
- RAM:      128 MiB DDR3
- Ethernet: 2x 10/100/1000 Mbps (switched, LAN + WAN)
- WIFI0:    MT7603E 2.4GHz 802.11b/g/n
- WIFI1:    MT7612E 5GHz 802.11ac
- Antennas: 2x internal, non-detachable
- LEDs:     Programmable LEDs: 5 blue LEDs (wlan, tel, sig1-3) and 2 red LEDs (wlan and sig1)
            Non-programmable "Power"  LED
- Buttons:  Reset and WPS

Instalation:
Install from TFTP

Set your PC IP to 10.10.10.3 and gateway to 10.10.10.123
Press "1" when turning on the router, and type the initramfs file name

You also need to solder pin header or cable to J4 or neighboring test points (T19-T21)
Pinouts from top to bottom: GND, TX, RX, VCC (3.3v)
Baudrate: 57600n8

There's also an additional gigabit transformer and RTL8211FD for the LTE module on the backside of the PCB.

Signed-off-by: Abdul Aziz Amar <abdulaziz.amar@gmail.com>

### CODE REVIEW

**Comment by kopijahe** [comment]
Board Pics:  
**TOP**
image

**BOTTOM**
image

**J4 and neighboring test points:**
image

**Comment by langit7** [comment]
> @mk248: Some guy in local facebook group said that the modem is connected through the "lan2" (port@3) with two different VLANs: 103 and 4094.
> 
> How do I add that to `02_network` file? Do I use `ucidef_add_switch_port`? I see it used on [WNDRMAC](https://github.com/openwrt/openwrt/blob/41be1a2de2f84f532a7eb1c11ece99811ca47ba5/target/linux/ath79/generic/base-files/etc/board.d/02_network#L386)...

Yes I have test  this scenario with put trunk interface to lan. LTE connection more stable compared to oem firmware 2x24 .

```
lan1.103  Link encap:Ethernet  HWaddr EE:9D:80:70:DE:35  
          inet addr:10.21.106.222  Bcast:255.255.255.255  Mask:255.255.255.255
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:7703274 errors:0 dropped:0 overruns:0 frame:0
          TX packets:4847379 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:12155863388 (11.3 GiB)  TX bytes:851860342 (812.3 MiB)

lan1.4094 Link encap:Ethernet  HWaddr EE:9D:80:70:DE:35  
          inet addr:169.254.0.2  Bcast:169.254.0.3  Mask:255.255.255.252
          inet6 addr: fe80::ec9d:80ff:fe70:de35/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:95334 errors:0 dropped:0 overruns:0 frame:0
          TX packets:68332 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:33485868 (31.9 MiB)  TX bytes:22143821 (21.1 MiB)

```
LTE itself is behind USB ethernet on top his firmware 
here bellow is take from oem partision mtd4  , part 0x9F823A - 0xC4D01A is modem firmware
```


DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
64            0x40            LZMA compressed data, properties: 0x5D, dictionary size: 33554432 bytes, uncompressed size: 14105984 bytes
10399898      0x9EB09A        CRC32 polynomial table, little endian
10432613      0x9F3065        Unix path: /home/bbwang/work/test/BOLT_ODU/SDK/work_BOLT/uboot
10453562      0x9F823A        uImage header, header size: 64 bytes, header CRC: 0x8F9B9C80, created: 2018-01-03 09:22:37, image size: 2444192 bytes, Data Address: 0xD0600000, Entry Point: 0xD0600000, data CRC: 0xCE2EFC3, OS: Linux, CPU: ARM, image type: OS Kernel Image, compression type: none, image name: "Linux-3.10.0-uc0"
10472014      0x9FCA4E        gzip compressed data, maximum compression, from Unix, last modified: 1970-01-01 00:00:00 (null date)
12898330      0xC4D01A        UBI erase count header, version: 1, EC: 0x0, VID header offset: 0x800, data offset: 0x1000

```

**Comment by langit7** [comment]
You can also flash without soldering with this step
```
Telnet to  router IP default is 192.168.1.1  
# cd /tmp/
# curl http://192.168.1.10/arion.bin -o arion.bin
# mtd_write -r write arion.bin Kernel
```
```
When i test this build i got a lot error when activating 5G wifi
[  635.677299] ieee80211 phy1: Hardware restart was requested
[  636.713123] mt76x2e 0000:02:00.0: MCU message 02 (seq 3) timed out
[  636.870270] mt76x2e 0000:02:00.0: Firmware Version: 0.0.00
[  636.881274] mt76x2e 0000:02:00.0: Build: 1
[  636.889455] mt76x2e 0000:02:00.0: Build Time: 201507311614____
[  636.904642] mt76x2e 0000:02:00.0: Firmware running!
[  636.917315] ieee80211 phy1: Hardware restart was requested
[  637.961130] mt76x2e 0000:02:00.0: MCU message 02 (seq 6) timed out
[  638.110266] mt76x2e 0000:02:00.0: Firmware Version: 0.0.00
[  638.121250] mt76x2e 0000:02:00.0: Build: 1
[  638.129468] mt76x2e 0000:02:00.0: Build Time: 201507311614____
[  638.144497] mt76x2e 0000:02:00.0: Firmware running!
[  638.157302] ieee80211 phy1: Hardware restart was requested
[  639.177125] mt76x2e 0000:02:00.0: MCU message 02 (seq 9) timed out
[  639.246248] mt76x2e 0000:02:00.0: Firmware Version: 0.0.00
[  639.257413] mt76x2e 0000:02:00.0: Build: 1
[  639.265592] mt76x2e 0000:02:00.0: Build Time: 201507311614____
[  639.280618] mt76x2e 0000:02:00.0: Firmware running!
```

**Comment by kopijahe** [comment]
@langit7: I don't think `telnetd` is enabled on stock firmware? At least not on B008 and B014 version.

**Comment by langit7** [comment]
> @langit7: I don't think `telnetd` is enabled on stock firmware? At least not on B008 and B014 version.
> 
> Can you provide the content of `/etc/config/network` after setting up the modem?

it's enbaled on my unit
```
# nvram_get  TELNETD_ENABLE
1
```
```
root@OpenWrt:~# cat /etc/config/network 

config interface 'loopback'
        option device 'lo'
        option proto 'static'
        option ipaddr '127.0.0.1'
        option netmask '255.0.0.0'

config globals 'globals'
        option packet_steering '1'
        option ula_prefix 'fd88:cc71:59ca::/48'

config device
        option name 'br-lan'
        option type 'bridge'
        option ipv6 '0'

config interface 'lan'
        option device 'br-lan'
        option proto 'static'
        option netmask '255.255.255.0'
        option ip6assign '60'
        option ipaddr '192.168.8.1'

config interface 'wan'
        option device 'wan'
        option proto 'dhcp'

config interface 'wan6'
        option device 'wan'
        option proto 'dhcpv6'

config interface 'wwan'
        option proto 'dhcp'
        option auto '0'

config interface 'eth_om'
        option proto 'static'
        option ipaddr '169.254.0.2'
        option netmask '255.255.255.252'
        option defaultroute '0'
        option device 'lan1.4094'

config device
        option name 'lan1'
        option ipv6 '0'

config device
        option type '8021q'
        option ifname 'lan1'
        option vid '103'
        option name 'lan1.103'
        option ipv6 '0'

config interface 'eth_data'
        option proto 'static'
        option device 'lan1.103'
        option netmask '255.255.255.255'
        option ipaddr '10.21.106.222'
        list dns '192.168.137.30'

config device
        option type '8021q'
        option ifname 'lan1'
        option vid '4094'
        option name 'lan1.4094'
```

File: target/linux/ramips/mt7621/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -33,6 +33,7 @@ ramips_setup_interfaces()
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
 		;;
 	asiarf,ap7621-001|\
+        bolt,arion|\
```
Comment (abajk) [inline target/linux/ramips/mt7621/base-files/etc/board.d/02_network:36]: Wrong indentation

**Comment by kopijahe** [comment]
> it's enbaled on my unit
> 
> ```
> # nvram_get  TELNETD_ENABLE
> 1
> ```

That's weird, I checked my devices and at least twoi of them does not have `telnetd` enabled.

File: target/linux/ramips/mt7621/base-files/etc/board.d/02_network (hunk 1 above)
Comment (kopijahe) [reply target/linux/ramips/mt7621/base-files/etc/board.d/02_network]: Ok, fixed in my newer commit.

**Comment by langit7** [comment]
diagram
I have no idea to implement this to dts and network config, without manual config

**Comment by kopijahe** [comment]
@langit7: If I did, how do I check that the modem is functional?

Is there any web interface from the modem? Or is there any IP that I can ping?

**Comment by langit7** [comment]
> @langit7: If I did, how do I check that the modem is functional?
> 
> Is there any web interface from the modem? Or is there any IP that I can ping?

To access to modem just access Web and telnet to 169.254.0.1, its just like direct ethernet trunk connected to port 0  with several vlan 100-107 and  4094 for image flash and management.

**Comment by langit7** [comment]
console port
ttl modem
Baudrate 921600

**Comment by radityabh** [comment]
> > @langit7: If I did, how do I check that the modem is functional?
> > Is there any web interface from the modem? Or is there any IP that I can ping?
> 
> To access to modem just access Web and telnet to 169.254.0.1, its just like direct ethernet trunk connected to port 0 with several vlan 100-107 and 4094 for image flash and management.



> console port ttl modem Baudrate 921600

does the module can at command in console? or just debug info

**Comment by langit7** [comment]
> 
> does the module can at command in console? or just debug info
yes, you can use attest or lted_cli
```
>at!gstatus

!GSTATUS:

Current Time : 0        Mode : ONLINE
System mode : LTE       PS state : Attached
LTE band : B40  LTE bw : 20MHz
LTE Rx chan : 38948     LTE Tx chan : 38948
EMM state : 0   RRC state : RRC IDLE
IMS reg state : No Srv
RSSI (dBm) : -55        Tx Power : -550
RSRP (dBm) : -86        TAC : 6(255)
RSRQ (dBm) : -10.0      Cell ID : 111203xxx
SINR (dB) : 25

OK

```

**Comment by radityabh** [comment]
> > does the module can at command in console? or just debug info
> > yes, you can use attest or lted_cli

ok man thanks. i dont get it, how modem can connect in openwrt with vlan mode. where you set, the internet setting? ex APN. can you explain?

**Comment by langit7** [comment]
> diagram I have no idea how to implement this to dts and network config, without manual config

Please have look  on the image above , there actually 2 routers , 1 router with gdm7432 on top mt7621 router. all lte stuff configured inside gdm7432 router. lte router connected to main router internally to lan port0 .

**Comment by radityabh** [comment]
> > diagram I have no idea how to implement this to dts and network config, without manual config
> 
> Please have look on the image above , there actually 2 routers , 1 router with gdm7432 on top mt7621 router. all lte stuff configured inside gdm7432 router. lte router connected to main router internally to lan port0 .

okey i get it. now, how i can up interface lan1 for lte modules? i use openwrt firmware from https://blog.kopijahe.my.id/posts/bolt-arion-openwrt/

**Comment by kopijahe** [comment]
@radityabh: that's the neat part, you can't (yet).

I'm trying to figure it out while working full-time. I'm already able to create the VLANs with uci-defaults, but still cannot ping or telnet to the modem.

**Comment by langit7** [comment]
some adjustment may needed 
```
  target/linux/ramips/mt7621/base-files/etc/board.d/02_network
        bolt,arion)
                ucidef_set_interfaces_wan  "lan1"
                ucidef_set_interfaces_lan_wan "lan2" "wan"
                ;;
```
target/linux/ramips/dts/mt7621_bolt_arion.dts 
```
&switch0 {
        ports {
                port@0 {
                        status = "okay";
                        label = "lan1";
                };
                wan: port@1{
                        status = "okay";
                        label = "wan";
                        nvmem-cells = <&macaddr_factory_4000>;
                        nvmem-cell-names = "mac-address";
                };

                port@2 {
                        status = "okay";
                        label = "lan2";
                };
        };
};
```
/etc/config/network 
```
config device
        option type '8021q'
        option ifname 'lan1'
        option vid '4094'
        option name 'lan1.4094'

config device
        option type '8021q'
        option ifname 'lan1'
        option vid '103'
        option name 'lan1.103'

config interface 'ethom'
        option device 'lan1.4094'
        option proto 'static'
        option ipaddr '169.254.0.2'

config interface 'ltedata'
        option device 'lan1.103'
        option proto 'static'
        option ipaddr '10.34.58.129'
        option netmask '255.255.255.255'

```

**Comment by kopijahe** [comment]
@langit7: ok, later this afternoon I'll implement this changes.

**Comment by radityabh** [comment]
@langit7 , after i add port@0 in dts. now i can ping to 169.254.0.1. but how i know ip for vlan vid 103? i try to use console in lte module, but the output is wierd 
image
have any suggest?

**Comment by langit7** [comment]
> Baudrate 921600  check /tmp/syslog for vlan generation . there is script  handled vlan population.

**Comment by kopijahe** [comment]
Ok, looks like I got it? I compiled busybox with telnet capability, so I can telnet from the router itself.

image

But I still cannot access the web interface of the modem...

**Comment by langit7** [comment]
> Ok, looks like I got it? I compiled busybox with telnet capability, so I can telnet from the router itself.
> 

> But I still cannot access the web interface of the modem...
169.254 is special ip you may try to set static route from your  pc to the modem.
in my win11 without this command bellow i will not able to connect .
`route add 169.254.0.1 mask 255.255.255.255 192.168.1.1
`
It is nice progress BTW,  could you please set eth_data Ips from telnet command output?

**Comment by radityabh** [comment]
> > Ok, looks like I got it? I compiled busybox with telnet capability, so I can telnet from the router itself.
> 
> > But I still cannot access the web interface of the modem...
> > 169.254 is special ip you may try to set static route from your  pc to the modem.
> > in my win11 without this command bellow i will not able to connect .
> > `route add 169.254.0.1 mask 255.255.255.255 192.168.1.1 `
> > It is nice progress BTW,  could you please set eth_data Ips from telnet command output?

I can access the web interface. but why I can't get signal ? 
image

**Comment by langit7** [comment]
make sure antenna connected and try do power cycle .

**Comment by kopijahe** [comment]
> It is nice progress BTW, could you please set eth_data Ips from telnet command output?

@langit7: this?

```
# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast master br0 state UP qlen 1000
    link/ether 00:0a:3b:f0:01:30 brd ff:ff:ff:ff:ff:ff
3: tunl0: <NOARP> mtu 0 qdisc noop state DOWN
    link/ipip 0.0.0.0 brd 0.0.0.0
4: sit0: <NOARP> mtu 1480 qdisc noop state DOWN
    link/sit 0.0.0.0 brd 0.0.0.0
5: lte0pdn0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:00 brd ff:ff:ff:ff:ff:ff
6: lte0pdn1: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:01 brd ff:ff:ff:ff:ff:ff
7: lte0pdn2: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:02 brd ff:ff:ff:ff:ff:ff
8: lte0pdn3: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN qlen 1000
    link/ether 00:0a:3b:f0:00:03 brd ff:ff:ff:ff:ff:ff
9: lte0pdn4: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:04 brd ff:ff:ff:ff:ff:ff
10: lte0pdn5: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:05 brd ff:ff:ff:ff:ff:ff
11: lte0pdn6: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:06 brd ff:ff:ff:ff:ff:ff
12: lte0pdn7: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:07 brd ff:ff:ff:ff:ff:ff
13: lte0pdn8: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:08 brd ff:ff:ff:ff:ff:ff
14: lte0pdn9: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:09 brd ff:ff:ff:ff:ff:ff
15: lte0pdn10: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:0a brd ff:ff:ff:ff:ff:ff
16: lte0pdn11: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:0b brd ff:ff:ff:ff:ff:ff
17: lte0pdn12: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:0c brd ff:ff:ff:ff:ff:ff
18: lte0pdn13: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:0d brd ff:ff:ff:ff:ff:ff
19: lte0pdn14: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:0e brd ff:ff:ff:ff:ff:ff
20: lte0pdn15: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether 00:0a:3b:f0:00:0f brd ff:ff:ff:ff:ff:ff
21: br0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP
    link/ether 00:0a:3b:f0:01:30 brd ff:ff:ff:ff:ff:ff
    inet 169.254.0.1/30 brd 169.254.0.3 scope global br0
       valid_lft forever preferred_lft forever
    inet6 fe80::20a:3bff:fef0:130/64 scope link
       valid_lft forever preferred_lft forever
22: eth_om@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br0 state UP
    link/ether 00:0a:3b:f0:01:30 brd ff:ff:ff:ff:ff:ff
```

I'm able to get "Connected" status and get some IP in the "device status" page, but there's still no internet...

image

**Comment by radityabh** [comment]
@kopijahe try IP from web UI, put in vid 103 IP. 
sad, I try many time my modem is not connected, do you have advice?

**Comment by langit7** [comment]
> @kopijahe try IP from web UI, put in vid 103 IP. sad, I try many time my modem is not connected, do you have advice?

make sure you  update dns and also add default gtw,
ip route add default dev modem.103

**Comment by radityabh** [comment]
> make sure you update dns and also add default gtw, ip route add default dev modem.103

no man, I still stuck in web modem not connected. 
image

**Comment by langit7** [comment]
> > make sure you update dns and also add default gtw, ip route add default dev modem.103
> 
> no man, I still stuck in web modem not connected. image

please check lte setting , set full band search and auto connection  set to always on

**Comment by radityabh** [comment]
> please check lte setting , set full band search and auto connection set to always on, or are you using Telkomsel? if yes you need to change IMEI first.

before I change to openwrt, I registered IMEI and work perfect in stock FW. And when set connection to always on, this error apear.
image

**Comment by langit7** [comment]
> > please check lte setting , set full band search and auto connection set to always on, or are you using Telkomsel? if yes you need to change IMEI first.
> 
> before I change to openwrt, I registered IMEI and work perfect in stock FW. And when set connection to always on, this error apear.
telnet to modem and reboot from there

**Comment by kopijahe** [comment]
> @kopijahe try IP from web UI, put in vid 103 IP.
> sad, I try many time my modem is not connected, do you have advice?

> make sure you update dns and also add default gtw,
> ip route add default dev modem.103

Ok, now I can ping 8.8.8.8 from openwrt using LTE.  
But dang, so complicated just to get it working.

Do I need to change the IP each time I want to use LTE?

**Comment by langit7** [comment]
I'm using shell script to check wan ip  ,  When i compare to stock  they using some own protocol and stored the wan ip  to nvram, unfortunately there is no documentation how its working. so for now im using shell script check every 5 sec to update ip/dns

**Comment by radityabh** [comment]
> > telnet to modem and reboot from there

its normal to wait this massage for long?
image

**Comment by langit7** [comment]
> its normal to wait this massage for long?
from my experience   its not normal ,  could you try manual attach using at command?

**Comment by radityabh** [comment]
> > its normal to wait this massage for long?
> > from my experience   its not normal ,  could you try manual attach using at command?

NO, I check in Process in telnet. is a process to command attach using at command, but I don't now from where trigger the process . I suspect my module maybe broke or version lte fw is diferrent.

**Comment by langit7** [comment]
you can try to boot with stock using tftpboot.

**Comment by radityabh** [comment]
> you can try to boot with stock using tftpboot.

yes and boom. no service in stock FW. hihi

**Comment by radityabh** [comment]
> from stock go to maintenance and do reset factory
nope, still no service. i think imei is blocked again.

**Comment by langit7** [comment]
If board still detected then it's may some config issue.

**Comment by kopijahe** [comment]
@radityabh: I was experiencing this issue earlier, I then solved it by resetting the LTE module itself.

**Comment by radityabh** [comment]
> @radityabh: I was experiencing this issue earlier, I then solved it by resetting the LTE module itself.

how? I try factory reset in module FW its not work

**Comment by langit7** [comment]
```
root@OpenWrt:~# lsusb  -t
/:  Bus 02.Port 1: Dev 1, Class=root_hub, Driver=xhci-mtk/1p, 5000M
/:  Bus 01.Port 1: Dev 1, Class=root_hub, Driver=xhci-mtk/2p, 480M
```
@kopijahe 
do you think we need to add kmod-usb3 kmod-ledtrig-usbdev ?

**Comment by kopijahe** [comment]
> how? I try factory reset in module FW its not work

Yes, factory reset from the LTE module firmware itself.

> do you think we need to add kmod-usb3 kmod-ledtrig-usbdev ?

I don't think so? I think it's not even enabled in the stock firmware, `lsusb` output from built-in tools does not return anything.

**Comment by kopijahe** [comment]
@langit7: I also tested with a static busybox, the USB is not activated in the stock firmware (there's also none available externally):

image

**Comment by langit7** [comment]
another option using curl to get lte ip , buts its still failed. 
```
#!/bin/bash

DATA=`curl -s http://169.254.0.1/cgi-bin/login.cgi?Command=getLoginStok -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36' -H 'Referer: http://169.254.0.1/Login.html'`
STOK=`echo "$DATA"| awk -F '[<>]' '/login_stok/{print $3 }'`
echo $STOK

LOGINX=`curl  -s http://169.254.0.1/cgi-bin/login.cgi -H "X-Stok: $STOK" -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36' -H 'Referer: http://169.254.0.1/Login.html' -H 'Origin: http://169.254.0.1' --data-raw 'Command=setUserCheck&input_URN=admin&rand_PWD=admin'`

SESS=`curl -c cookie.txt -vL -s http://169.254.0.1/cgi-bin/login.cgi -H "X-Stok: $STOK" -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36' -H 'Referer: http://169.254.0.1/Login.html' -H 'Origin: http://169.254.0.1' --data-raw 'Command=setLOGINvalue&input_URN=admin&rand_PWD=admin'`

RES=`curl  --trace-ascii -  -s -vL  'http://169.254.0.1/cgi-bin/status.cgi?Command=getLocalNetworkStatusASP' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0' -H 'Accept: */*' -H 'Accept-Language: en-US,en;q=0.5'  -H 'Connection: keep-alive' -H 'Referer: http://169.254.0.1/status.html' -b cookie.txt` 
```

**Comment by radityabh** [comment]
@langit7 thanks you man for your kindness. after I buy one device again. its works perfect.
image

for refresh IP I use telnet. to get IP address . its works in startup or schedule task.

**Comment by langit7** [comment]
Here  telnet version but some people may don't to mod busybox to support telnet for security reason.
https://github.com/langit7/Bolt-arion-pl100/blob/main/script/lteipchecker-telnet.sh

**Comment by Kacrut88** [comment]
@langit7 bro, pls, fw openwrt pl100 final still run no more setting include lte modem?? just install frm breed web

**Comment by langit7** [comment]
> @langit7 bro, pls, fw openwrt pl100 final still run no more setting include lte modem?? just install frm breed web

I'm still using stool bootloader you can try my build here https://github.com/langit7/Bolt-arion-pl100/releases

**Comment by Kacrut88** [comment]
> if i try y'r build, can i use flash frm breed web ???

**Comment by langit7** [comment]
>  if i try y'r build, can i use flash frm breed web ???

If you have openwrt better upgrade from luci or using sysupgrade, breed should be working also

**Comment by thecupangin** [comment]
> > @langit7 bro, pls, fw openwrt pl100 final still run no more setting include lte modem?? just install frm breed web
> 
> I'm still using stock bootloader you can try my build here https://github.com/langit7/Bolt-arion-pl100/releases

hi bro @langit7 i use  your build, i can telnet the modem ip but i cannot access the web gui ip, how come ya?

image

File: target/linux/ramips/mt7621/base-files/etc/board.d/02_network [hunk 2]
```diff
@@ -50,6 +50,11 @@ ramips_setup_interfaces()
 	xiaomi,mi-router-4a-gigabit)
 		ucidef_set_interfaces_lan_wan "lan1 lan2" "wan"
 		;;
+	bolt,arion)
+		ucidef_set_interfaces_lan_wan "lan" "wan"
+                ucidef_set_interface "eth_data" device "modem.103" protocol "static" ipaddr "10.22.127.222" netmask "255.255.255.255"
+                ucidef_set_interface "eth_om" device "modem.4094" protocol "static" ipaddr "169.254.0.2" netmask "255.255.255.252"
```
Comment (hauke) [inline target/linux/ramips/mt7621/base-files/etc/board.d/02_network:56]: Please fix the indention

File: target/linux/ramips/dts/mt7621_bolt_arion.dts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+&gmac0 {
+	nvmem-cells = <&macaddr_factory_4000>;
+	nvmem-cell-names = "mac-address";
+	mac-address-increment = <3>;
+};
+
+&switch0 {
+	ports {
+                port@0 {
```
Comment (hauke) [inline target/linux/ramips/dts/mt7621_bolt_arion.dts:147]: Please fix the indention

**Comment by langit7** [comment]
> > > @langit7 bro, pls, fw openwrt pl100 final still run no more setting include lte modem?? just install frm breed web
> > 
> > 
> > I'm still using stock bootloader you can try my build here https://github.com/langit7/Bolt-arion-pl100/releases
> 
> hi bro @langit7 i use your build, i can telnet the modem ip but i cannot access the web gui ip, how come ya?
> 
are you pinging from router or  pc ?   if web keep loading  you   need to edit  /sbin/lte and replace with code  here 
https://raw.githubusercontent.com/langit7/Bolt-arion-pl100/main/script/lteipchecker-telnet.sh.
try add static route from your pc to router ip
`route add 169.254.0.1 mask 255.255.255.255 192.168.1.1`

**Comment by robbyaprianto** [comment]
> > > > @langit7 bro, pls, fw openwrt pl100 final still run no more setting include lte modem?? just install frm breed web
> > > 
> > > 
> > > I'm still using stock bootloader you can try my build here https://github.com/langit7/Bolt-arion-pl100/releases
> > 
> > 
> > hi bro @langit7 i use your build, i can telnet the modem ip but i cannot access the web gui ip, how come ya?
> 
> are you pinging from router or pc ? if web keep loading you need to edit /sbin/lte and replace with code here https://raw.githubusercontent.com/langit7/Bolt-arion-pl100/main/script/lteipchecker-telnet.sh. try add static route from your pc to router ip `route add 169.254.0.1 mask 255.255.255.255 192.168.1.1`

Ok, i can access now, but there is some problem again. First time i can login, but after reboot, i cannot login again into web gui

**Comment by langit7** [comment]
> Ok, i can access now, but there is some problem again. First time i can login, but after reboot, i cannot login again into web 
@robbyaprianto ,
- access the router using ssh and check if any process called /sbin/lte  and terminate the process
- edit /sbin/lte file  replace with content bellow 
```
#!/bin/sh
while sleep 5; do
{ echo "nvram_get cache wan_ipaddr"; echo "nvram_get cache wan_primary_dns"; sleep 2; } | telnet 169.254.0.1|grep -Eo '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}'|grep -v "169.254.0.1"  > /tmp/ipmodem.txt

ethdataip=`ifconfig modem.103 | grep 'inet addr' | cut -d: -f2 | awk '{print $1}'`
lteip=`head -1 /tmp/ipmodem.txt`
ltedns=`tail -1 /tmp/ipmodem.txt`
if [ -z "$lteip" ]
then
      echo "Ip modem tidak dapat di cek"
else
     if [[ "$ethdataip" != "$lteip" ]];then
        echo " eth_data $ethdataip \n LTE IP $lteip dns $ltedns"
         #ifconfig modem.103 $lteip netmask 255.255.255.255 up
        logger -p notice -t lte "IP berubah dar $ethdata menjadi $lteip dns $ltedns"
         uci set network.eth_data.ipaddr="$lteip"
         uci set network.eth_data.dns="$ltedns"
         uci commit network
         /etc/init.d/network reload
         ip route add default dev modem.103
	 sleep 60
        else
         echo "OK IP eth_data $ethdataip"
        fi
fi
done
```

File: target/linux/ramips/mt7621/base-files/etc/board.d/02_network (hunk 2 above)
Comment (kopijahe) [reply target/linux/ramips/mt7621/base-files/etc/board.d/02_network]: Ok fixed.

File: target/linux/ramips/dts/mt7621_bolt_arion.dts (hunk 3 above)
Comment (kopijahe) [reply target/linux/ramips/dts/mt7621_bolt_arion.dts]: Ok fixed.

**Comment by radityabh** [comment]
@langit7 do you know command to check 2CA active or not? I think, its not active. I check with different modem in same place and use 2CA (B40+B40). 
image

**Comment by mans0n** [comment]
So it seems most things are now settled down, could you please squash your changes into a single commit?
https://openwrt.org/docs/guide-developer/working-with-github-pr#squashing_commits
And please do not create a merge commit, just do rebase on top of the master branch.

**Comment by kopijahe** [comment]
I accidentaly pressed the merge button, sorry ._.

**Comment by hauke** [comment]
Thank you for your patch, I applied it to master in 78c3534645958c123aa82cec9926a34eed5dd5dd.
